Sources/SwiftParser/Lexer.swift

```diff
@@ -1104,11 +1240,10 @@ extension Lexer.Cursor {
           // The only case we reach here is unterminated single line string in the
           // interpolation. For better recovery, go on after emitting an error.
           //          diagnose(CurPtr, diag::lex_unterminated_string)
-          wasErroneous = true
           continue
         } else {
           //          diagnose(TokStart, diag::lex_unterminated_string)
-          return (.unknown, [])
+          return LexerResult(.stringLiteralContents, newMode: .normal)
```

Comment:
This change is dangerous without possibly corresponding changes to string literal contents parsing. We need to make sure that its scanning and indexing logic can tolerate an early eof and strings like `"\(xyz \v"`

Reply:
It seems to be working… At least the existing test cases didn't catch anything. I haven't banged it harder than that though